### PR TITLE
Add RenameTestCases transformer

### LINKED ISSUE #190: Allow to escape : in transformer parameters
Character `:` is used as parameter delimeter:
```
robotidy --configure Transformer:param=value:param2=value
```
But if value of paramer should contain `:` it will throw error expecting the second parameter:
```
robotidy --configure Transformer:param=:
```
Allow to escape `:` with `\`:
robotidy --configure Transformer:param=\:

### PR DESCRIPTION
**Pull request by bhirsz**
I've decided to split #183 into test case and keyword names.

I've also removed unnecessary characters from class descriptions and added url to transformer detailed docs to each description.

Closes #190

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ Contribution guideline
 
 Getting started
 ----------------
-Did you spot bug or you really want to see some feature implemented? It's always best to start with create issue in github.
+Did you spot a bug or you really want to see some feature implemented? It's always best to start with creating issue in GitHub.
```
Comment (bhirsz) [inline CONTRIBUTING.md:6]: Somehow I missed two comments from previous PR

File: CHANGES.md [hunk 2]
```diff
@@ -5,6 +5,8 @@
 ### Features
 - It is now possible to provide source paths in configuration file [#154](https://github.com/MarketSquare/robotframework-tidy/issues/154)
 - Non default transformers can be enabled using ``enabled=True`` parameter [#182](https://github.com/MarketSquare/robotframework-tidy/issues/182)
+- New non default transformer `RenameTestCases`. It capitalizes first letter of the test case name, remove trailing dot and can replace provided regex pattern with substitute string [#183](https://github.com/MarketSquare/robotframework-tidy/issues/183)
```
Comment (mnojek) [inline CHANGES.md:8]:
```suggestion
- New non default transformer `RenameTestCases`. It capitalizes first letter of the test case name, removes trailing dot and can replace provided regex pattern with substitute string [#183](https://github.com/MarketSquare/robotframework-tidy/issues/183)
```

File: CHANGES.md [hunk 3]
```diff
@@ -5,6 +5,8 @@
...
+- Semicolon in parameter value can be now escaped with `\:` [#190](https://github.com/MarketSquare/robotframework-tidy/issues/190)
```
Comment (mnojek) [inline CHANGES.md:9]:
```suggestion
- Semicolon in parameter value can now be escaped with `\:` [#190](https://github.com/MarketSquare/robotframework-tidy/issues/190)
```

File: docs/source/transformers/RenameTestCases.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
+.. _RenameTestCases:
+
+RenameTestCases
+================================
+
+Enforce test case naming.
+
+RenameTestCases is not included in default transformers that's why you need to call it with ``--transform`` explicitly::
```
Comment (mnojek) [inline docs/source/transformers/RenameTestCases.rst:8]:
```suggestion
RenameTestCases is not included in default transformers, that's why you need to call it with ``--transform`` explicitly::
```

File: docs/source/transformers/RenameTestCases.rst [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    robotidy --transform RenameTestCases src
+
+Or configure `enable` parameter::
+
+    robotidy --configure RenameTestCases:enabled=True
+
+This transformer capitalize first letter of test case name, remove trailing dot and strip leading/trailing whitespace.
```
Comment (mnojek) [inline docs/source/transformers/RenameTestCases.rst:16]:
```suggestion
This transformer capitalizes first letter of test case name, removes trailing dot and strips leading/trailing whitespace.
```

File: docs/source/transformers/RenameTestCases.rst [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+It is also possible to configure `replace_pattern` parameter to find and replace regex pattern. Use `replace_to`
+to set replacement value. This configuration::
```
Comment (mnojek) [inline docs/source/transformers/RenameTestCases.rst:19]:
```suggestion
to set a replacement value. This configuration::
```

File: robotidy/transformers/NormalizeSectionHeaderName.py [hunk 7]
```diff
@@ -9,13 +9,13 @@
...
     Normalize section headers names.
-    Robot Framework is quite flexible with the section header naming. Following lines are equal::
+    Robot Framework is quite flexible with the section header naming. Following lines are equal:
 
         *setting
         *** SETTINGS
         *** SettingS ***
 
-    This transformer normalize naming to follow ``*** SectionName ***`` format (with plurar variant)::
+    This transformer normalize naming to follow ``*** SectionName ***`` format (with plurar variant):
```
Comment (mnojek) [inline robotidy/transformers/NormalizeSectionHeaderName.py:18]:
```suggestion
    This transformer normalizes naming to follow ``*** SectionName ***`` format (with plural variant):
```

File: robotidy/transformers/OrderSettings.py [hunk 8]
```diff
@@ -14,7 +14,7 @@ class OrderSettings(ModelTransformer):
     Order settings like [Arguments], [Setup], [Return] inside Keywords and Test Cases.
 
     Keyword settings [Documentation], [Tags], [Timeout], [Arguments] are put before keyword body and
-    settings like [Teardown], [Return] are moved to the end of keyword::
+    settings like [Teardown], [Return] are moved to the end of keyword:
```
Comment (mnojek) [inline robotidy/transformers/OrderSettings.py:17]:
```suggestion
    settings like [Teardown], [Return] are moved to the end of the keyword:
```

File: robotidy/transformers/RenameTestCases.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    See https://robotidy.readthedocs.io/en/latest/transformers/RenameTestCases.html for more examples.
+    """
+    def __init__(self, replace_pattern: Optional[str] = None, replace_to: Optional[str] = None):
+        try:
+            self.replace_pattern = re.compile(replace_pattern) if replace_pattern is not None else None
+        except re.error as err:
+            raise click.BadOptionUsage(
+                option_name='transform',
+                message=f"Invalid configurable value: '{replace_pattern}' for replace_pattern in RenameTestCases"
+                        f" transformer. It should be valid regex expression. Regex error: '{err.msg}'")
```
Comment (mnojek) [inline robotidy/transformers/RenameTestCases.py:46]:
```suggestion
                        f" transformer. It should be a valid regex expression. Regex error: '{err.msg}'")
```

**Review by mnojek — APPROVED** [review]

**Pull request merged**